### PR TITLE
Add CreateFromJson POST operation to ScaleUnit

### PR DESCRIPTION
**Pull request by cednation**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json [hunk 1]
```diff
@@ -19,14 +19,14 @@
     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Fabric.Admin/fabricLocations/{location}/scaleUnits/{scaleUnit}/ScaleOut": {
```
Comment (martabarillas) [inline specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: 'ScaleOut' should be 'scaleOut'

File: specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/examples/ScaleUnit/CreateFromJson.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+    "parameters": {
+        "api-version": "2016-05-01",
+        "subscriptionId": "20ff7fc3-e762-44dd-bd96-b71116dcdc23",
+        "resourceGroupName": "System.local",
+        "location": "local",
```
Comment (martabarillas) [inline specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/examples/ScaleUnit/CreateFromJson.json]: Missing 'scaleUnit' parameter

File: specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/examples/ScaleUnit/CreateFromJson.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        "creationData": {
+            "clusterName": "cluster",
```
Comment (martabarillas) [inline specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/examples/ScaleUnit/CreateFromJson.json]: wouldn't clusterName and ScaleUnit refer to the same thing?

File: specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json [hunk 4]
```diff
@@ -288,6 +331,71 @@
...
+      "properties": {
+        "clusterName": {
+          "description": "Cluster name for the new scale unit.",
+          "type": "string"
+        },
+        "physicalNodes": {
+          "description": "List of nodes in the scale unit.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DeploymentJsonPhysicalNodeParameters"
```
Comment (martabarillas) [inline specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: Definition of DeploymentJsonPhysicalNodeParameters is missing

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6243 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6243/commits/ecaee494ff72e8b71c30774e08d8ee5b7097c4fc

**Comment by martabarillas** [comment]
adding @bganapa for review

File: specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json [hunk 5]
```diff
@@ -352,6 +352,21 @@
         }
       }
     },
+    "DeploymentJsonPhysicalNodeParameters": {
+      "description": "Description of a bare metal node used for scale unit scale out operations.",
+      "type": "object",
+      "properties": {
+        "bmcIpAddress": {
```
Comment (martabarillas) [inline specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: bmcIpAddress or bmcIPv4Address? the label is different than in the sample

File: specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json [hunk 6]
```diff
@@ -288,6 +331,86 @@
...
+          "description": "Computer name of the physical machine.",
+          "type": "string"
+        }
+      }
+    },
+    "CreateFromJsonScaleUnitParametersList": {
+      "description": "A list of input data that allows for creating the new scale unit.",
+      "type": "object",
+      "properties": {
+        "clusterName": {
```
Comment (martabarillas) [inline specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: 'clusterName' got removed from the sample bc is the same as the scaleUnit name...should be removed from here?

Comment (cednation) [reply specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: No and I think it is wrong for me to remove it from the sample, I will put it back. I've looked over the code again to try to figure it out. (I did not write this code, it was Lincoln). The customer can call in two different ways, either supplying the name or not supplying the name.

Comment (cednation) [reply specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: To supply the name it is a POST operation on ..\scaleUnit\<name>  and the name is in the body data. The code only uses the name in the body, the name on the URL is not used.
To create a cluster where a name is chosen by default, user does a POST on ..\scaleUnit. In that case the clusterName in the body should be null, although we don't check for that.

Comment (martabarillas) [reply specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: sounds good

Comment (cednation) [reply specification/azsadmin/resource-manager/fabric/Microsoft.Fabric.Admin/preview/2016-05-01/ScaleUnit.json]: Never mind... digging through the code more the clusterName parameter in the body does not get used... originally it looked like it did. So not sure why it was defined. I've removed it from the Swagger and example.

**Review by bganapa — APPROVED** [review]

**Comment by bganapa** [comment]
@salameer Could you please help assigning this to one of the Spec reviewers? It looks like it did not auto-assign. Thanks!

**Review by martabarillas — APPROVED** [review]

**Comment by bganapa** [comment]
@sphibbs Could you please assign to one of the spec reviewers? It did not auto assign for some reason. Thanks!

**Review by tjprescott — APPROVED** [review]

**Pull request merged**